accessibility/README.md

```diff
@@ -0,0 +1,495 @@
+
+#### Exceptions
+* You should consider using a `<a>` element in cases where clicking on the element updates the URL and visiting that URL directly would display a different page
+* An example would be Colour Swatches, where clicking on a swatch would update the current page and load in a new product in Vue, but the data is fetched from a separate product with its own URL
+* The benefit here is that web crawlers can read these links to discover the other pages
+
+```vue
+<a
+  :href="`/products/${swatchProduct.handle}`"
+  @click.prevent="handleSwatchClick"
+  v-text="swatchProduct.color"
+/>
+```
+
+### Use input events rather than wrapping click events
+
+#### Don't
+
+```vue
+<li
+  v-for="item in items"
+  :key="item.id"
+  @click="filter(item.value)"
+>
+  <input
+    :id="item.id"
+    :checked="item.active"
+    type="checkbox"
+    :value="item.value"
+  >
+  <label
+    :for="item.id"
+    v-text="item.label"
+  >
+</li>
+```
+
+#### Do
+
+```vue
+<li
+  v-for="item in items"
+  :key="item.id"
+>
+  <input
+    :id="item.id"
+    :checked="item.active"
+    type="checkbox"
+    :value="item.value" 
+    @change="filter($event.target.value)"
+  >
+
+  <label :for="item.id" v-text="item.label" />
+</li>
+```
+
+* Instead of adding a click event to an `input` element's parent, use the native `input` or `change` events
+
+[ꜛ Back to TOC](#table-of-contents)
+
+## Keyboard Support
+
+* [Test keyboard focus](#test-keyboard-focus)
+* [Test keyboard tab order](#test-keyboard-tab-order)
+* [Focus Trap](#focus-trap)
+
+### Test keyboard focus
+* Ensure that interactive elements can be focussed and activated using a keyboard
+* Ensure that functionality that supports mouse interactions (click, mousedown, mouseleave etc) have suitable keyboard-accessible alternatives
+
+### Test keyboard tab order
+* Ensure that keyboard focus order is correct
+  * The order that items are focussed when tabbing should match the visual order of the elements on the page
+  * Try an ensure that the order elements in the DOM match the visual order on the site
+* Exceptions can be made here when the order changes responsively, e.g. when the order is correct on mobile screen sizes 
+
+### Focus Trap
+
+* Use a focus trap when opening overlays so that a keyboard user is only able to tab between elements inside the overlay
+* Ensure that the focus trap is cleared when an overlay is closed
+* Ensure that the focus trap is updated when the overlay state changes
+  * Example: if you have a multi-tiered navigation, ensure that the focus trap updates as the user navigates between tiers
+  * Example: if you have a search overlay and the results update as you type, ensure that focus trap updates as the content changes
+* Focus trapping can be achieved in Canvas projects with the `focusTrap` and `previousElement` helpers in the `accessibility.js` store - see [Canvas Gitbook - accessibility.js](https://we-make-websites.gitbook.io/canvas/features/vuex-stores/accessibility.js)
```

Comment:
Frame 3 documentation is kept separate in the _frame/_ folder. By default this documentation assumes you're talking about a Canvas project.

The fact that most of the documentation is written in Vue suggests that Frame 3 could be ignored here.